overlord/servicestate/servicemgr.go

```diff
@@ -42,8 +54,224 @@ func Manager(st *state.State, runner *state.TaskRunner) *ServiceManager {
 	return m
 }
 
+// inactiveEnterTimestampForUnit returns the time that a given unit entered the
+// inactive state as defined by systemd docs. Specifically this time is the most
+// recent time in which the unit transitioned from deactivating ("Stopping") to
+// dead ("Stopped"). It may be the zero time if this has never happened during
+// the current boot, since this property is only tracked during the current
+// boot. It specifically does not return a time that is monotonic, so the time
+// returned here may be subject to bugs if there was a discontinuous time jump
+// on the system before or during the unit's transition to inactive.
+// XXX: move to systemd package ?
+func inactiveEnterTimestampForUnit(unit string) (time.Time, error) {
+	// XXX: ignore stderr of systemctl command to avoid further infractions
+	//      around LP #1885597
+	out, err := exec.Command("systemctl", "show", "--property", "InactiveEnterTimestamp", unit).Output()
```

Comment:
hmm, not sure you're reading it properly, it's specifically the `InactiveEnterTimestamp` not `InactiveEnterTimestampMonotonic` being used here...

Reply:
But also, I completely agree that we don't have a way to get the "monotonic timestamp" of a file modification, so hence it is better to use the non-monotonic version of `InactiveEnterTimestamp` when we ask systemd for the time to get better bounds. I also agree in practice that time jumps are very unlikely to happen now, but we shouldn't discard the possibility, since snapd refreshes may happen very shortly after a time change if a device gets access to network and sync's its time right before a refresh happens, so we should keep it in mind, but since we have many other bugs surrounding time jumps and refreshes, etc I think we are safe to ignore it for now, but when we have a way to handle time jumps, we should handle it here too if possible, hence the comment calling out that it currently doesn't work with time jumps